Lib/test/test_capi/test_watchers.py

```diff
@@ -336,6 +336,102 @@ def test_no_more_ids_available(self):
                 self.add_watcher()
 
 
+class TestCodeObjectWatchers(unittest.TestCase):
+    @contextmanager
+    def code_watcher(self, which_watcher):
+        wid = _testcapi.register_code_watcher(which_watcher)
+        try:
+            yield wid
+        finally:
+            _testcapi.clear_code_watcher(wid)
+
+    def make_counts_dict(self, created_1, destroyed_1, created_2, destroyed_2):
+        return {
+            0: {"created": created_1, "destroyed": destroyed_1},
+            1: {"created": created_2, "destroyed": destroyed_2},
+        }
+
+    def test_code_object_events_dispatched(self):
+        # verify that all counts are zero before any watchers are registered
+        self.assertDictEqual(
+            self.make_counts_dict(0, 0, 0, 0),
+            _testcapi.get_code_watchers_events_counts(),
+        )
```

Comment:
nit: this is kind of asking for another helper assert method (e.g. `assertCodeWatcherCounts`) that just takes the four counts and doesn't require repeating `_testcapi.get_code_watchers_events_counts()` every time.